File: Basics/Hash-Map/hash.py
class HashMap:
    def __init__(self):
        self.map = {}

    def insert(self, key, value):
        self.map[key] = value

    def retrieve(self, key):
        return self.map.get(key)

# Example usage:
my_hash_map = HashMap()
my_hash_map.insert("key1", "value1")
my_hash_map.insert("key2", "value2")

print(my_hash_map.retrieve("key1"))  # Output: value1
print(my_hash_map.retrieve("key2"))  # Output: value2

File: Basics/Hash-Map/hash.js
class HashMap {
    constructor() {
      this.map = {};
    }
  
    insert(key, value) {
      this.map[key] = value;
    }
  
    retrieve(key) {
      return this.map[key];
    }
  }
  
  // Example usage:
  const myHashMap = new HashMap();
  myHashMap.insert("key1", "value1");
  myHashMap.insert("key2", "value2");
  
  console.log(myHashMap.retrieve("key1")); // Output: value1
  console.log(myHashMap.retrieve("key2")); // Output: value2
  